sklearn/metrics/_plot/cap_curve.py

```diff
@@ -0,0 +1,274 @@
+        y_true_sorted = y_bool[sorted_indices]
+        sample_weight_sorted = sample_weight[sorted_indices]
+
+        # compute cumulative sums for true positives and all cases
+        cumulative_true_positives = np.cumsum(y_true_sorted * sample_weight_sorted)
+        cumulative_total = np.cumsum(sample_weight_sorted)
+
+        viz = cls(
+            cumulative_true_positives=cumulative_true_positives,
+            cumulative_total=cumulative_total,
+            estimator_name=name,
+            pos_label=pos_label,
+        )
+
+        return viz.plot(ax=ax, name=name, **kwargs)
+    
+
+    @classmethod
+    def from_estimator(
+        cls,
+        estimator,
+        X,
+        y,
+        *,
+        sample_weight=None,
+        response_method="auto",
+        pos_label=None,
+        name=None,
+        ax=None,
+        **kwargs,
+    ):
+        """Create a CAP Curve display from an estimator.
+
+        Parameters
+        ----------
+        estimator : estimator instance
+            Fitted classifier or a fitted :class:`~sklearn.pipeline.Pipeline`
+            in which the last estimator is a classifier.
+
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Input values.
+
+        y : array-like of shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Sample weights.
+
+        response_method : {'predict_proba', 'decision_function', 'auto'} \
+                default='auto'
+            Specifies whether to use :term:`predict_proba` or
+            :term:`decision_function` as the target response. If set to 'auto',
+            :term:`predict_proba` is tried first and if it does not exist
+            :term:`decision_function` is tried next.
+
+        pos_label : int, float, bool or str, default=None
+            The class considered as the positive class when computing metrics.
+            By default, `estimators.classes_[1]` is considered as the positive class.
+
+        name : str, default=None
+            Name of CAP Curve for labeling. If `None`, use the name of the estimator.
+
+        ax : matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is created.
+
+        **kwargs : dict
+            Keyword arguments to be passed to matplotlib's `plot`.
+
+        Returns
+        -------
+        display : :class:`~sklearn.metrics.CumulativeAccuracyDisplay`
+            The CAP Curve display.
+        """
+        
+        # validate and prepare the prediction scores
+        if response_method == "auto":
+            if hasattr(estimator, "predict_proba"):
+                response_method = "predict_proba"
+            elif hasattr(estimator, "decision_function"):
+                response_method = "decision_function"
+            else:
+                raise ValueError("Estimator does not have a predict_proba or decision_function method.")
```

Comment:
We should allow also accept regressors, as long as all the predicted values are positive (and raise `ValueError` otherwise).

You might want to use `from sklearn.base import is_classifier` or `is_regressor` to tell whether an estimator is a classifier or a regressor (in which case the response method to use should be `predict`).

Reply:
To be done in a follow PR (see https://github.com/scikit-learn/scikit-learn/pull/28752#issuecomment-2039611884).